packages/@webex/plugin-cc/src/AgentProfileService/AgentProfileService.ts

```diff
@@ -0,0 +1,146 @@
+} from './types';
+
+export default class AgentProfileService {
+  ciUserId: string;
+  orgId: string;
+  webex: WebexSDK;
+  wccAPIURL: string;
+
+  constructor(ciUserId: string, orgId: string, webex: WebexSDK, wccAPIURL: string) {
+    this.ciUserId = ciUserId;
+    this.orgId = orgId;
+    this.webex = webex;
+    this.wccAPIURL = wccAPIURL;
+  }
+
+  /**
+   * Method to get User using CI.
+   * @param {string} ciUserId The CI Id of the user.
+   * @param {string} orgId The ID of an Organization.
+   * @returns {Promise<UserResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getUserUsingCI(ciUserId: string, orgId: string): Promise<UserResponse> {
+    try {
+      if (!ciUserId || !orgId) Promise.reject(new Error('ciUserId or orgId is undefined'));
+
+      const URL = `${this.wccAPIURL}organization/${orgId}/user/by-ci-user-id/${ciUserId}`;
+      this.webex.logger.log('getUserUsingCI api called successfully.');
+
+      return Promise.resolve(await this.makeGETRequest(URL));
+    } catch (error) {
+      this.webex.logger.error(`Error while calling getUserUsingCI API, ${error}`);
+
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Method to get Desktop Profile by passing Org Id.
+   * @param {string} orgId The ID of an Organization.
+   * @param {string} desktopProfileId ID of the Desktop Profile to be retrieved.
+   * @returns {Promise<DesktopProfileResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async retrieveDesktopProfileById(
+    orgId: string,
+    desktopProfileId: string
+  ): Promise<DesktopProfileResponse> {
+    try {
+      if (!orgId || !desktopProfileId)
+        Promise.reject(new Error('orgId or desktopProfileId is undefined'));
+
+      const URL = `${this.wccAPIURL}organization/${orgId}/agent-profile/${desktopProfileId}`;
+
+      return Promise.resolve(await this.makeGETRequest(URL));
+    } catch (error) {
+      this.webex.logger.error(`Error while calling retrieveDesktopProfileById API, ${error}`);
+
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Method to get List of Teams.
+   * @param {string} orgId The ID of an Organization.
+   * @param {number} page (Optional) Index of the page of results to be fetched.
+   * @param {number} pageSize (Optional) Number of items to be displayed on a page.
+   * @param {Array<String>} filter (Optional) Filter which can be applied to the elements to be fetched.
+   * @param {Array<String>} attributes (Optional) Specify the attributes to be returned. Supported attributes are id, name, active and workTypeCode.
+   * @returns {Promise<ListTeamsResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getListOfTeams(
+    orgId: string,
+    page?: number,
+    pageSize?: number,
+    filter?: string[],
+    attributes?: string[]
+  ): Promise<ListTeamsResponse> {
+    try {
+      if (!orgId) Promise.reject(new Error('orgId is undefined'));
```

Comment:
is this if condition really required? Is there a chance of getListofTeams getting called with empty orgId? When does that happen?

Reply:
We are fetching the orgId using this code:- const orgId = await this.webex.credentials.getOrgId(); 
It might be the case where the getOrgId function returns undefined, with undefined it will make the API URL Invalid. I believe it is not a good practice to hit the API Server with invalid URL with orgId as undefined. This should be taken care at application level. So, I introduced this check to avoid hitting API server.

Reply:
So the question here is will we ever get undefined from `this.webex.credentials.getOrgId()`? can we check if there can be a token without an org id?
Additionally, can we check if the orgId is already present in any other object inside the `webex` object instead of credentials?

Reply:
@pagour98 have we checked if we can get the orgId from any other part of the `webex` object?

Reply:
@sreenara logic has been updated. Now I am fetching orgId from webex.internal.device object.